core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/container/postgres/PostgresConfigService.java

```diff
@@ -60,15 +61,30 @@ public class PostgresConfigService {
     @Inject
     private UpgradeRdsBackupRestoreStateParamsProvider upgradeRdsBackupRestoreStateParamsProvider;
 
+    @Inject
+    private ClusterService clusterService;
+
     public void decorateServicePillarWithPostgresIfNeeded(Map<String, SaltPillarProperties> servicePillar, StackDto stackDto) {
         Map<String, Object> postgresConfig = initPostgresConfig(stackDto);
-
-        RedbeamsDbSslDetails sslDetails = dbCertificateProvider.getRelatedSslCerts(stackDto);
-        Set<String> rootCerts = sslDetails.getSslCerts();
-        if (CollectionUtils.isNotEmpty(rootCerts)) {
-            Map<String, String> rootSslCertsMap = Map.of("ssl_certs", String.join("\n", rootCerts),
-                    "ssl_certs_file_path", dbCertificateProvider.getSslCertsFilePath(),
-                    "ssl_enabled", String.valueOf(sslDetails.isSslEnabledForStack()));
+        String rootCertsBundle;
+        boolean sslEnabled;
+        boolean restartRequired;
```

Comment:
this could be a nice pojo with a `toMap` method which would generate the `rootSslCertsMap`